rpc-rs/src/otel.rs

```diff
@@ -1,139 +1,54 @@
+#![cfg(not(target_arch = "wasm"))]
 //! Contains helpers and code for enabling [OpenTelemetry](https://opentelemetry.io/) tracing for
 //! wasmbus-rpc calls. Please note that right now this is only supported for providers. This module
 //! is only available with the `otel` feature enabled
 
-use std::collections::HashSet;
-
-/// NOTE: The commented out code in here will work once we upgrade to async_nats. I have left it so that we don't have to rewrite it again
-// use async_nats::header::HeaderName;
-// use async_nats::HeaderMap;
-use nats_aflowt::header::HeaderMap;
-use opentelemetry::propagation::{Extractor, Injector, TextMapPropagator};
-use opentelemetry::sdk::propagation::TraceContextPropagator;
+use async_nats::header::HeaderMap;
+#[cfg(feature = "otel")]
+use opentelemetry::{
+    propagation::{Extractor, Injector, TextMapPropagator},
+    sdk::propagation::TraceContextPropagator,
+};
+#[cfg(feature = "otel")]
 use tracing::span::Span;
+#[cfg(feature = "otel")]
 use tracing_opentelemetry::OpenTelemetrySpanExt;
 
+#[cfg(feature = "otel")]
+macro_rules! span_record {
+    ($span:ident, $name:expr, $val:expr) => {
+        $span.record($name, &tracing::field::display($val));
+    };
+}
+#[cfg(not(feature = "otel"))]
```

Comment:
Wouldn't we want the fields anyway for logging?

Reply:
we only create the span if otel is enabled, so we can't use span.record if otel is disabled.